agent/lib_aws_sdk_php.c

```diff
@@ -519,6 +518,171 @@ char* nr_lib_aws_sdk_php_get_command_arg_value(char* command_arg_name,
+        this_obj, base_class, "endpoint");
+    if (nr_php_is_zval_valid_object(endpoint_zval)) {
+      host_zval = nr_php_get_zval_object_property(endpoint_zval, "host");
+      if (nr_php_is_zval_non_empty_string(host_zval)) {
+        datastore_params->instance->host = Z_STRVAL_P(host_zval);
+
+        /* Only try to get a port if we have a valid host. */
+        port_zval = nr_php_get_zval_object_property(endpoint_zval, "port");
+        if (nr_php_is_zval_valid_integer(port_zval)) {
+          /* Must be freed by caller */
+          datastore_params->instance->port_path_or_id
+              = nr_formatf(NR_INT64_FMT, Z_LVAL_P(port_zval));
+        } else {
+          /* In case where host was found but port was not, spec says return
+           * unknown for port. */
+          datastore_params->instance->port_path_or_id = nr_strdup("unknown");
+        }
+      }
+    }
+  }
+  if (NULL == datastore_params->instance->host) {
+    /* Unable to retrieve the endpoint, go with AWS defaults. */
+    datastore_params->instance->host = AWS_SDK_PHP_DYNAMODBCLIENT_DEFAULT_HOST;
+    /* Need to strdup because the calling function will free it. */
+    datastore_params->instance->port_path_or_id
+        = nr_strdup(AWS_SDK_PHP_DYNAMODBCLIENT_DEFAULT_PORT);
+  }
+
+  table_name = nr_lib_aws_sdk_php_get_command_arg_value(
+      AWS_SDK_PHP_DYNAMODBCLIENT_TABLENAME_ARG, NR_EXECUTE_ORIG_ARGS);
+  if (!nr_strempty(table_name)) {
+    /* Must be freed by caller */
+    datastore_params->collection = table_name;
+  }
+  if (!nr_strempty(NRINI(aws_account_id))) {
+    account_id = NRINI(aws_account_id);
+  }
+
+  if (NULL != datastore_params->collection && NULL != account_id
+      && NULL != cloud_attrs->cloud_region) {
+    /* Must be freed by caller */
+    cloud_attrs->cloud_resource_id = nr_formatf(
+        "arn:aws:dynamodb:%s:%s:table/%s", cloud_attrs->cloud_region,
+        account_id, datastore_params->collection);
+  }
+}
+
+void nr_lib_aws_sdk_php_dynamodb_handle(nr_segment_t* auto_segment,
+                                        char* command_name_string,
+                                        size_t command_name_len,
+                                        NR_EXECUTE_PROTO) {
+  nr_segment_t* datastore_segment = NULL;
+  nr_segment_cloud_attrs_t cloud_attrs = {0};
+  nr_datastore_instance_t instance = {0};
+  nr_segment_datastore_params_t datastore_params = {
+    .db_system = AWS_SDK_PHP_DYNAMODBCLIENT_DATASTORE_SYSTEM,
+    .datastore = {
+      .type = NR_DATASTORE_DYNAMODB,
+    },
+    .instance = &instance,
+    .callbacks = {
+      .backtrace = nr_php_backtrace_callback,
+    },
+  };
+  if (NULL == auto_segment) {
+    return;
+  }
+
+  if (NULL == command_name_string || 0 == command_name_len) {
+    return;
+  }
+
+#define AWS_COMMAND_IS(CMD) \
+  (command_name_len == (sizeof(CMD) - 1) && nr_streq(CMD, command_name_string))
+
+  /* Determine if we instrument this command. */
+  if (AWS_COMMAND_IS("createTable")) {
```

Comment:
There are defines for all of these strings - can we use them instead of having the literal value duplicated in two places?

Reply:
There are no defines for these strings.

You'll note that these [02c00c9](https://github.com/newrelic/newrelic-php-agent/pull/1037/commits/02c00c96d55cbddc1e8b71c2cfcc3b56f343bba7) are determined by spec so that all agents report the same string regardless of a particular language's syntax.
For PHP, the aws-sdk-php command is `createTable` but we need to report `create_table` which are two different strings.